src/test/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppSchedulerTests.java

```diff
@@ -0,0 +1,368 @@
+import io.pivotal.scheduler.v1.ExpressionType;
+import io.pivotal.scheduler.v1.calls.Calls;
+import io.pivotal.scheduler.v1.jobs.CreateJobRequest;
+import io.pivotal.scheduler.v1.jobs.CreateJobResponse;
+import io.pivotal.scheduler.v1.jobs.DeleteJobRequest;
+import io.pivotal.scheduler.v1.jobs.DeleteJobScheduleRequest;
+import io.pivotal.scheduler.v1.jobs.ExecuteJobRequest;
+import io.pivotal.scheduler.v1.jobs.ExecuteJobResponse;
+import io.pivotal.scheduler.v1.jobs.GetJobRequest;
+import io.pivotal.scheduler.v1.jobs.GetJobResponse;
+import io.pivotal.scheduler.v1.jobs.JobResource;
+import io.pivotal.scheduler.v1.jobs.JobScheduleResource;
+import io.pivotal.scheduler.v1.jobs.Jobs;
+import io.pivotal.scheduler.v1.jobs.ListJobHistoriesRequest;
+import io.pivotal.scheduler.v1.jobs.ListJobHistoriesResponse;
+import io.pivotal.scheduler.v1.jobs.ListJobScheduleHistoriesRequest;
+import io.pivotal.scheduler.v1.jobs.ListJobScheduleHistoriesResponse;
+import io.pivotal.scheduler.v1.jobs.ListJobSchedulesRequest;
+import io.pivotal.scheduler.v1.jobs.ListJobSchedulesResponse;
+import io.pivotal.scheduler.v1.jobs.ListJobsRequest;
+import io.pivotal.scheduler.v1.jobs.ListJobsResponse;
+import io.pivotal.scheduler.v1.jobs.ScheduleJobRequest;
+import io.pivotal.scheduler.v1.jobs.ScheduleJobResponse;
+import org.cloudfoundry.client.CloudFoundryClient;
+import org.cloudfoundry.client.v2.applications.ApplicationsV2;
+import org.cloudfoundry.client.v3.tasks.Tasks;
+import org.cloudfoundry.operations.CloudFoundryOperations;
+import org.cloudfoundry.operations.applications.ApplicationSummary;
+import org.cloudfoundry.operations.applications.Applications;
+import org.cloudfoundry.operations.spaces.SpaceSummary;
+import org.cloudfoundry.operations.spaces.Spaces;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties;
+import org.springframework.cloud.scheduler.spi.core.ScheduleInfo;
+import org.springframework.cloud.scheduler.spi.core.SchedulerException;
+import org.springframework.cloud.scheduler.spi.core.SchedulerPropertyKeys;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+/**
+ * Test the core features of the Spring Cloud Scheduler implementation.
+ *
+ * @author Glenn Renfro
+ */
+public class CloudFoundryAppSchedulerTests {
+
+	public static final String DEFAULT_CRON_EXPRESSION = "* * * * *";
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private Applications applications;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private CloudFoundryOperations operations;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private Spaces spaces;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private ApplicationsV2 applicationsV2;
```

Comment:
Why does this use V2?  V2 doesn't even know of the concept of a task...

Reply:
done